File: model/ConfigData.cs
using System;
using Newtonsoft.Json;

namespace Com.GitHub.ZachDeibert.CodeTanks.Model {
    public sealed class ConfigData {
        public sealed class CostData {
            [JsonProperty("motor")]
            public Formula MotorCost;
            [JsonProperty("battery")]
            public Formula BatteryCost;
            [JsonProperty("cannon")]
            public Formula CannonCost;
            [JsonProperty("radar")]
            public Formula RadarCost;
            [JsonProperty("gps")]
            public Formula GPSCost;
            [JsonProperty("explosives")]
            public Formula ExplosivesCost;
        }

        public sealed class SpeedData {
            [JsonProperty("bullet")]
            public Formula BulletSpeed;
            [JsonProperty("tankDrive")]
            public Formula TankDriveSpeed;
            [JsonProperty("tankTurn")]
            public Formula TankTurnSpeed;
        }

        public sealed class ExplosionData {
            [JsonProperty("size")]
            public Formula Size;
            [JsonProperty("duration")]
            public int Duration;
        }

        [JsonProperty("funds")]
        public double TotalFunds;
        [JsonProperty("costs")]
        public CostData Costs;
        [JsonProperty("speeds")]
        public SpeedData Speeds;
        [JsonProperty("explosions")]
        public ExplosionData Explosions;
        [JsonProperty("tankSize")]
        public double TankSize;
    }
}

File: apis/java/src/main/java/com/github/zachdeibert/codetanks/api/model/ConfigData.java
package com.github.zachdeibert.codetanks.api.model;

public final class ConfigData {
	public static final class CostData {
		public Formula motor;
		public Formula battery;
		public Formula cannon;
		public Formula radar;
		public Formula gps;
		public Formula explosives;
	}

	public static final class SpeedData {
		public Formula bullet;
		public Formula tankDrive;
		public Formula tankTurn;
	}

	public static final class ExplosionData {
		public Formula size;
		public int duration;
	}

	public double funds;
	public CostData costs;
	public SpeedData speeds;
	public ExplosionData explosions;
	public double tankSize;
}
